precompiles/avs/tx.go

```diff
@@ -61,24 +69,126 @@ func (p Precompile) RegisterOrDeregisterAVSInfo(
 	return method.Outputs.Pack(true)
 }
 
-func (p Precompile) BindOperatorToAVS(
+func (p Precompile) DeregisterAVS(
 	ctx sdk.Context,
 	_ common.Address,
 	contract *vm.Contract,
 	_ vm.StateDB,
 	method *abi.Method,
 	args []interface{},
 ) ([]byte, error) {
-	if len(args) != 2 {
-		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 2, len(args))
+	if len(args) != len(p.ABI.Methods[MethodDeregisterAVS].Inputs) {
+		return nil, xerrors.Errorf(cmn.ErrInvalidNumberOfArgs, len(p.ABI.Methods[MethodDeregisterAVS].Inputs), len(args))
+	}
+	avsParams := &avskeeper.AVSRegisterOrDeregisterParams{}
+
+	avsName, ok := args[0].(string)
+	if !ok || avsName == "" {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 0, "string", avsName)
+	}
+	avsParams.AvsName = avsName
+
+	avsAddress, err := util.ProcessAddress(contract.Address().String())
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "parse avsAddress error")
+	}
+	avsParams.AvsAddress = avsAddress
+
+	callerAddress, err := util.ProcessAddress(contract.CallerAddress.String())
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "parse callerAddress error")
+	}
+	avsParams.CallerAddress = callerAddress
+
+	avsParams.Action = avskeeper.DeRegisterAction
+
+	if err != nil {
+		return nil, err
+	}
+	err = p.avsKeeper.AVSInfoUpdate(ctx, avsParams)
+	if err != nil {
+		return nil, err
+	}
+	return method.Outputs.Pack(true)
+}
+
+func (p Precompile) UpdateAVS(
+	ctx sdk.Context,
+	_ common.Address,
+	contract *vm.Contract,
+	_ vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	// parse the avs input params first.
+	avsParams, err := p.GetAVSParamsFromUpdateInputs(ctx, args)
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "parse args error")
+	}
+	avsAddress, err := util.ProcessAddress(contract.Address().String())
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "parse avsAddress error")
 	}
+	avsParams.AvsAddress = avsAddress
+
+	callerAddress, err := util.ProcessAddress(contract.CallerAddress.String())
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "parse callerAddress error")
+	}
+	avsParams.CallerAddress = callerAddress
+
+	avsParams.Action = avskeeper.UpdateAction
+
+	if err != nil {
+		return nil, err
+	}
+	err = p.avsKeeper.AVSInfoUpdate(ctx, avsParams)
+	if err != nil {
+		return nil, err
+	}
+	return method.Outputs.Pack(true)
+}
+
+func (p Precompile) BindOperatorToAVS(
+	ctx sdk.Context,
+	_ common.Address,
+	contract *vm.Contract,
+	_ vm.StateDB,
+	method *abi.Method,
+	_ []interface{},
+) ([]byte, error) {
 	operatorParams := &avskeeper.OperatorOptParams{}
-	action, ok := args[0].(uint64)
-	if !ok || (action != avskeeper.RegisterAction && action != avskeeper.DeRegisterAction) {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 0, "uint64", action)
+
+	callerAddress, err := util.ProcessAddress(contract.CallerAddress.String())
+	if err != nil {
```

Comment:
Same here, to check the permission firstly

Reply:
Permission has been checked in keep：if !k.operatorKeeper.IsOperator(ctx, opAccAddr)